torch/_inductor/dependencies.py

```diff
@@ -220,10 +193,18 @@ def canonicalize(
 
         # assign new variables each dimension to deal with numbering mismatches
         # d0, d1, d2 could become d0, d2 -- which won't match d0, d1
-        _, add_var = var_builder(canonicalization_prefix())
+        new_vars, add_var = var_builder(canonicalization_prefix())
         replacement = dict(zip(index_vars, reindex([add_var(x) for x in new_sizes])))
-
         index = sympy_subs(sympy.expand(index), replacement)
+
+        new_vars = [*new_vars.keys()]
+        new_sizes = [*new_sizes]
+        free_symbols = index.free_symbols
+        while new_vars and new_vars[-1] not in free_symbols:
+            # Reduction has last (reduced) dim in its sizes, but
```

Comment:
are there cases where downstream users still would have reduced dim? I remember I needed both versions to enable more fusions.

Reply:
I believe this logic will remove the extra dim from both the producer and the downstream users.